### PR TITLE
Added Stress Level Detection Model

### PR DESCRIPTION
**Pull request by J-B-Mugundh**
Description:
This pull request introduces a new feature for detecting mental stress levels based on social media interactions using a Random Forest model. The implementation includes the following changes:

Model Integration: Integrated the trained random_forest_model.joblib to predict stress levels based on user inputs.

Prediction Logic: Implemented a function to assess stress levels and provide tailored advice based on the predicted value:

- Low Stress (less than 1.5): Suggests maintaining a balanced lifestyle.
- Moderate Stress (between 1.5 and 3.5): Recommends taking breaks and practicing stress relief techniques.
- High Stress (greater than 3.5): Advises seeking professional help or implementing stress management strategies.

Testing:
The feature has been tested locally, ensuring that predictions return accurate stress levels and corresponding advice based on various input scenarios.

Next Steps:
Review and provide feedback on the implementation.
Consider any additional features or improvements for user experience.

Related Issue:
#25

Screenshots: 

image
image 

Thank you for reviewing this PR!

### CODE REVIEW

**Comment by J-B-Mugundh** [comment]
@yashasvini121 Please review the PR and let me know if it needs any improvements. Thank you 😊

**Review by yashasvini121 — CHANGES_REQUESTED** [review]
The code is totally fine, but some minor changes would improve it further. Sync your fork before making any updates as well.

File: models/stress_level_detect/model.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+from joblib import load
+
+# Load the trained Random Forest model
+model = load('models/stress_level_detect/random_forest_model.joblib')
```
[the remark this answers is not in the document]
Comment (J-B-Mugundh) [reply models/stress_level_detect/model.py:4]: > * You have used a pre-trained model. If you trained it yourself, please add the Jupyter notebook or include the training process in the `model.py` file.
> * Additionally, if you have the dataset, please include that as well.
> * If you are using a pre-trained model from an external source, please mention the source of the model. Any references or URLs in a comment would help.

I'll add the notebook file where I trained.

File: models/stress_level_detect/predict.py
[the remark this answers is not in the document]
Comment (J-B-Mugundh) [reply models/stress_level_detect/predict.py:1]: > This file lacks the `model_details()` function. **If possible**, please add the model evaluation or provide a model description that includes its source, any relevant references, and additional details.
> 
> If this requires updates in the `page_handler.py`, you can leave it as is or make the necessary fixes.

Will it be fine if I can add the graphs which I have in notebook like?

image

[the remark this answers is not in the document]
Comment (yashasvini121) [reply models/stress_level_detect/predict.py:1]: Yes, you can include graphs, but please only add those that directly explain the problem statement, as this particular graph isn't very helpful. Alternatively, if your updates aren't general to all models, you could create a custom class from `PageHandler` in `Stress_Level_Detector.py` and add your required functions there.

Additionally, if you have a useful and relevant notebook, please add it to the `notebooks/` folder.

[the remark this answers is not in the document]
Comment (J-B-Mugundh) [reply models/stress_level_detect/predict.py:1]: I don't find much useful graphs around here!

**Comment by J-B-Mugundh** [comment]
@yashasvini121 Is it fine now or should I add or change anything more?

**Review by yashasvini121 — APPROVED** [review]

**Comment by yashasvini121** [comment]
One final piece of feedback: avoid working directly on the master branch. Instead, create a new branch for each issue to prevent potential merge conflicts.

**Comment by J-B-Mugundh** [comment]
> Great!! One final piece of feedback: avoid working directly on the master branch. Instead, create a new branch for each issue to prevent potential merge conflicts.

Sure! I hadn't thought of that.. Thanks so much for bringing it up! 😊

**Pull request merged**